packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +44,349 @@ import {Heading, Text} from '@react-spectrum/typography';
+    <Dialog aria-labelledby="heading-id">
+      <Text slot="header" id="heading-id">Terms and Conditions</Text> 
+      <Divider />
+      <Content>
+        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Mi proin sed libero enim. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Sed enim ut sem viverra aliquet eget sit amet tellus. Diam quis enim lobortis scelerisque fermentum dui faucibus in ornare. Diam quam nulla porttitor massa id. Eleifend mi in nulla posuere sollicitudin. Turpis nunc eget lorem dolor sed viverra ipsum nunc. Faucibus in ornare quam viverra. Risus commodo viverra maecenas accumsan lacus vel facilisis volutpat est. Nam libero justo laoreet sit amet cursus sit. Netus et malesuada fames ac. Dictum fusce ut placerat orci nulla pellentesque dignissim enim sit. Eros donec ac odio tempor orci. Ut etiam sit amet nisl purus in mollis nunc. Nisl rhoncus mattis rhoncus urna neque viverra. Convallis aenean et tortor at risus. Diam phasellus vestibulum lorem sed risus ultricies.</Text>
+      </Content>
+      <ButtonGroup>
+        <Button autoFocus variant="cta" onPress={close}>Acknowledge</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
 ## Events
 
-*If the component supports event props, include this section. Only cover the events that are important to the main functionality of the component.*
+For Dialogs, user defined callbacks should be chained with the DialogTrigger's `close` function in the press/click handler
+of the Dialog's action buttons. The following example alerts if the Dialog's save or cancel button is clicked.
+
+```tsx example
+function Example() {
+  let alertSave = (cb) => {
+    cb();
+    alert('Profile saved!');
+  }
+
+  let alertCancel = (cb) => {
+    cb();
+    alert('Profile not saved!');
+  }
+
+  return (
+    <DialogTrigger>
+      <ActionButton>Set Profile</ActionButton>
+      {(close) => (
+        <Dialog>
+          <Heading>Profile</Heading>
+          <Divider />
+          <ButtonGroup>
+            <Button autoFocus variant="cta" onPress={() => alertSave(close)}>Save</Button>
+            <Button variant="secondary" onPress={() => alertCancel(close)}>Cancel</Button>
+          </ButtonGroup>
+          <Content>
+            <Form>
+              <TextField label="Name" />
+              <Checkbox>Make private</Checkbox>
+            </Form>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Additionally, DialogTrigger accepts an `onOpenChange` prop which is triggered whenever the Dialog is opened or closed. 
+For more information on the `onOpenChange` prop see [DialogTrigger page](TODO LINK HERE).
 
-## Validation
+### Dismissable Dialogs
 
-*If the component supports validation props, include this section.*
+Dismissable Dialogs support a `onDismiss` prop which is triggered whenever the Dialog's close button is clicked.
```

Comment:
I talked to Daniel about how a user might expect this to be called on any dismiss event, including the `DialogTrigger` events like escape key or clicking on the overlay. He pointed out that users can use `onOpenChange` to catch these.